streaming/index.js

```diff
@@ -622,40 +622,63 @@ const startServer = async () => {
 
     log.verbose(req.requestId, `Starting stream from ${ids.join(', ')} for ${accountId}`);
 
-    // Currently message is of type string, soon it'll be Record<string, any>
-    const listener = message => {
-      const { event, payload, queued_at } = message;
+    const transmit = (event, raw_payload, filter_results) => {
+      let payload = raw_payload;
+      if ((event === 'update' || event === 'status.update') && filter_results && filter_results.length > 0) {
+        // copy the payload and add the `filtered` attribute, an Array of FilterResult
+        // https://docs.joinmastodon.org/entities/Status/#filtered
+        payload = {
+          ...payload,
+          filtered: filter_results
+        };
+      }
```

Comment:
I'm not sure what the advantage of doing that here is, since we're already having the `event === 'update' || event === 'status.update'` logic in the place we are computing `filter_results` and passing the payload…

Reply:
You'll still need the event name for logging, as this function moved up in scope, but yeah, I guess now that the code is more documented, doing the `{ ...payload, filtered: filter_results }` at transmit call-site is better. Will update